pycbc/waveform/ringdown.py

```diff
@@ -65,11 +64,47 @@ def props(obj, required, **kwargs):
 
     return input_params
 
+def format_lmns(lmns):
+    """ Checks if the format of the parameter lmns is correct, returning the
+    appropriate format if not, and raise an error if nmodes=0
+    """
+    # Case 0: the lmns is in the right format
+    if len(lmns[0]) == 3:
+        lmns = lmns
+
+    else:
+        # Case 1: the list is in a string with or without comma,
+        # "[221, 331]" or "[221 331]"
+        if lmns[0] == '[' and lmns[-1] == ']':
+            lmns = lmns.strip('[]')
+
+        # Case 2: a list with only one string with or without comma,
+        # ["221' '331"] or ["221', '331"]
+        elif len(lmns[0]) > 3:
+            lmns = lmns[0]
+
+        # Check if there is a comma or not
+        if bool(re.search(',', lmns)):
+            lmns = lmns.split(',')
+        else:
+            lmns = lmns.split()
+
+        # There may still be extra spaces or quotes, loop over items
+        lmns = [lmns[n].strip().strip('\'') for n in range(len(lmns))]
+
+    # Check that nmodes != 0
+    for lmn in lmns:
+        if int(lmn[2]) == 0:
+            raise ValueError('Number of overtones (nmodes) must be greater '
+                             'than zero.')
+
+    return lmns
```

Comment:
It's good to have this in a function. But I'm a bit confused by the allowed formats. Adding a doc string will help. Still, I'm worried that you're allowing for too many different formats, which can potentially lead to bugs. It would be best to only allow a list of strings of the modes, i.e., `['221', '331']`, raising an error if you get anything else. Are you allowing for these different formats because of the issues with passing in the list of strings from condor-parsed config files?

Reply:
Yes, these two different case-examples came from the inference workflows. Now the "Case 2" is what has been causing our inference workflows to ignore the second mode, but in the past we had also seen the "Case 1" (see https://github.com/gwastro/pycbc/pull/994 ). It seemed easier to add this function here, to account for those two cases we have observed from inference runs, rather than figuring out how to force configparser to keep this parameter as a list of string. I can add an extra line raising an error if the format is different than the cases considered here, in case new formats pop up in the future and we get wrong results again...

Reply:
Yeah, one case not caught by this is if there is any extra strings on the first item, then you end up dropping the additional modes. For example,
```
>>> format_lmns(['221 ',  '331'])
['221']
```
How about this: separate out the case when you get a string using `isinstance`, and do all the string parsing there. Then use `int` to ensure that you have the right format. Also, drop allowing space-separated things, since it just complicates things. I think it's ok to enforce that the modes be comma-separated. So, something like:
```
def format_lmns(lmns):
    if isinstance(lmns, (str, unicode)):
        # strip off brackets
        lmns = lmns.replace('[', '').replace(']', '')
        # convert to list of strings
        lmns = lmns.split(',')
    out = []
    # cycle over the lmns, ensure that we get back a list of strings that are
    # 3 digits long
    for lmn in lmns:
        if isinstance(lmn, (str, unicode)):
            # remove any trailing quotes
            lmn = lmn.replace('"', '').replace("'", "")
        # try to convert to int then str, to ensure right format
        lmn = str(int(lmn))
        if len(lmn) != 3:
            raise ValueError('Format of parameter lmns not recognized. See '
                             'approximant documentation for more info.')
        out.append(lmn)
    return out
```

Reply:
Ok, I will take care of this later today. Just one note: the workflow config parser was returning a string *without comma*, so I would keep the non-comma options

Reply:
@cdcapano I have now implemented your changes, though still allowing for the lmns to be separate only by a space and not by a comma (because of what I said in my previous comment). I also added a `try: ... except: ...` condition in Case 2 because it was failing when there was only one mode given but with an extra space (for instance, ['221 ']). This was, of course, because of the split by a space (which your code did not have, so it would work with your code I think)